### PR TITLE
[fix](planner)the common type of floating and decimal should be floating type

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java [hunk 1]
```diff
@@ -1077,12 +1077,26 @@ protected static Optional<DataType> findCommonPrimitiveTypeForCaseWhen(DataType
             return Optional.empty();
         }
 
+        // decimalv3 and floating type
+        if (t1.isDecimalV3Type() || t2.isDecimalV3Type()) {
+            if (t1.isFloatType() || t2.isDoubleType() || t1.isDoubleType() || t2.isFloatType()) {
+                return t1.isDecimalV3Type() ? Optional.of(t2) : Optional.of(t1);
+            }
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java:1085]: we need to select double or float according to decimal precision and scale

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**